components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java

```diff
@@ -0,0 +1,269 @@
+            throw handleServerException(ERROR_CODE_ERROR_CHECKING_DB_METADATA, e);
+        }
+    }
+
+    /**
+     * Get the tenant ID.
+     *
+     * @return the tenant ID.
+     */
+    public static int getTenantId() {
+
+        return PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+    }
+
+    /**
+     * Get the tenant domain.
+     *
+     * @return the tenant domain.
+     */
+    public static String getTenantDomain() {
+
+        return PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+    }
+
+    /**
+     * Get the organization id from context.
+     *
+     * @return the organization id.
+     */
+    public static String getOrganizationId() {
+
+        return PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId();
+    }
+
+    /**
+     * Get the username of the authenticated user.
+     *
+     * @return the username of the authenticated user.
+     */
+    public static String getAuthenticatedUsername() {
+
+        return PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+    }
+
+    /**
+     * Get the user ID.
+     *
+     * @return the user ID.
+     */
+    public static String getUserId() {
+
+        return PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserId();
+    }
+
+    /**
+     * Build URI prepending the server API context with the proxy context path to the endpoint.
+     *
+     * @param organizationId The organization ID.
+     * @return Relative URI.
+     */
+    public static String buildURIForBody(String organizationId) {
+
+        String context = getContext(V1_API_PATH_COMPONENT + PATH_SEPARATOR + ORGANIZATION_PATH
+                + PATH_SEPARATOR + organizationId);
+
+        return context;
+    }
+
+    /**
+     * Builds the API context.
+     *
+     * @param endpoint Relative endpoint path.
+     * @return Context of the API.
+     */
+    public static String getContext(String endpoint) {
+
+        String organizationId = getOrganizationId();
+        if (StringUtils.isNotBlank(organizationId)) {
+            return String.format(ORGANIZATION_CONTEXT_PATH_COMPONENT, organizationId) + SERVER_API_PATH_COMPONENT +
+                    endpoint;
+        } else {
+            return SERVER_API_PATH_COMPONENT + endpoint;
+        }
+    }
+
+    /**
+     * Generate unique identifier for the organization.
+     *
+     * @return organization id.
+     */
+    public static String generateUniqueID() {
+
+        return UUID.randomUUID().toString();
+    }
+
+    /**
+     * Create a List containing allowed permissions.
+     *
+     * @param resourceId permission provided
+     * @return List of allowed permissions.
+     */
+    public static List<String> getAllowedPermissions(String resourceId) {
+
+        String[] permissionParts = resourceId.split(PATH_SEPARATOR);
+        List<String> allowedPermissions = new ArrayList<>();
+        for (int i = 0; i < permissionParts.length - 1; i++) {
```

Comment:
unclear why we are not looping the last element `permissionParts.length - 1`. If it's valid, can you add a comment in the code explaining?

Reply:
comment added via 6534bc4a65f7a5c7396abda24169fb0e5690702a